### PR TITLE
Add '-x' flag which to extract code

### PR DESCRIPTION
**Pull request by josephsnyder**
Add a '-x' flag which will utilize the ViViaN generation process but will
leave the execution after the export of the routines from the VistA
instance.  This directory can then be copied out of a container after it
has been initalized.

### CODE REVIEW

**Comment by josephsnyder** [comment]
Other documentation found here: http://review.code.osehra.org/#/c/1049/1

File: ViViaN/vivianInstall.sh
Comment (ChristopherEdwards) [inline ViViaN/vivianInstall.sh]: Why does extracting need another argument? aka -x somethingHere. It wasn't used on the main script.

File: README.md
Comment (ChristopherEdwards) [inline README.md]: not sure that all autoInstaller flags should be here. It is just one more place that needs to be updated if more options are added in the future. Help is generated by the autoinstaller script. maybe just reference that section.

Though the specific option for doing extraction should be called out in this document - How to extract M components from a CACHE.DAT file.

**Comment by shabiel** [comment]
Joe Snyder,

Should I wait for you to make updates based on Chris' feedback or start
testing it now?

--Sam

**Comment by josephsnyder** [comment]
I'll be making fixes soon.  Please hold off until then.

- Joe

Joseph Snyder
R&D Engineer

Kitware, Inc.
28 Corporate Dr
Clifton Park, NY 12065
(518) 881-4409

**Review by ChristopherEdwards — APPROVED** [review]

**Comment by shabiel** [comment]
I spoke with Joe. The main issue right now is that you cannot run this
without a cache key. I want us to be able to do the extraction on a single
user version.

Also, need to add these lines after the example:

    docker cp cache:/opt/VistA-M /tmp/ # no need to put a cp -r
    docker stop cache
    docker rm cache
    docker rmi cachevista

**Comment by ChristopherEdwards** [comment]
You may want to use a volume to mount the cache-files directory instead of a copy into the container. then the files will be on your file system and you can then use the docker --rm flag.

**Comment by josephsnyder** [comment]
Hi @ChristopherEdwards and @shabiel ,

The recent updates are just a rebase after the merging of the GT.M enhancements.  PTAL again to make sure your approval still stands.  Thanks!

**Comment by ChristopherEdwards** [comment]
Approval still stands after rebase.

**Pull request merged**
